src/main/java/net/datafaker/providers/healthcare/Disease.java

```diff
@@ -0,0 +1,113 @@
+package net.datafaker.providers.healthcare;
+
+import net.datafaker.providers.base.AbstractProvider;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Generate random, different kinds of disease.
+ *
+ * @since 0.8.0
+ */
+public class Disease extends AbstractProvider<HealthcareProviders> {
+
+    static final String INTERNAL_DISEASE_KEY = "healthcare.disease.internal_disease";
+    static final String NEUROLOGICAL_DISEASE_KEY = "healthcare.disease.neurology";
+    static final String SURGICAL_DISEASE_KEY = "healthcare.disease.surgery";
+    static final String PAEDIATRIC_DISEASE_KEY = "healthcare.disease.paediatrics";
+    static final String GYNECOLOGY_AND_OBSTETRICS_DISEASE_KEY = "healthcare.disease.gynecology_and_obstetrics";
+    static final String OPHTHALMOLOGY_AND_OTORHINOLARYNGOLOGY_DISEASE_KEY = "healthcare.disease.ophthalmology_and_otorhinolaryngology";
+    static final String DERMATOLOGY_DISEASE_KEY = "healthcare.disease.dermatology";
```

Comment:
Probably it's better to use enum for this

Reply:
I don't mind if you keep it as is, these are mostly (only?) internal things anyway, we can always refactor it later.

Reply:
if you use enum, then you don't need `private final List<Supplier<String>> allDiseaseProviders` at all
and `anyDisease` could be implemented as `Options.option..`

And each time there will be a new decease, then with current approach you need to add noth string var for this decease + maintain `allDiseaseProviders`. With enums you just need to update enums...
Thus finally less code

Reply:
Hm, I need to play with that, cause `Options.option` is new to me. Thanks for that hint, sounds reasonable now.

Reply:
Hi all, hope you don't mind, but I've addressed these comments in a commit on this PR, I didn't want to let these efforts go to waste. Thanks @filipowm for this improvement. @snuyanzin , can you please have another look at the PR?